main.c

```diff
@@ -232,6 +253,24 @@ void cleanup_after_vm(supervisor_allocation* heap) {
     #endif
     reset_board();
     reset_status_led();
+
+    // Save next code info.
+    if (next_code) {
+        if (next_code_allocation) {
+            free_memory(next_code_allocation);
+        }
+        // empty filename and options 0 is equivalent to the default, no need to waste memory on that
+        if (next_code_len > 1 || next_code->options != 0) {
+            // TODO do I need to increase CIRCUITPY_SUPERVISOR_ALLOC_COUNT for this?
```

Comment:
I don't think so. It is set to 8 and we have: stack, heap, flash cache, terminal tilegrid, sharp display, rgb display at most.

Reply:
I see 12 in https://github.com/adafruit/circuitpython/blob/dd86cb00adb2ecb1e0cdf256d9a4e0f96a0c6de1/supervisor/shared/memory.c#L34 But yeah, then that should be even more sufficient.

Reply:
I’m not so sure about that anymore – on closer examination, the rgb display needs 4 allocations and can occur CIRCUITPY_DISPLAY_LIMIT times, the sharp display too, so I get

```
rgbmatrix     4*2
sharpdisplay  1*2
usbmidi       1
terminal      1
flash cache   1
stack         1
heap          1
              =15
```

Although maybe no board has all these features enabled at the same time (the only one with CIRCUITPY_DISPLAY_LIMIT = 2 I can see is monster_m4sk).

Let’s not worry about it for now as I may end up making CIRCUITPY_SUPERVISOR_ALLOC_COUNT depend on enabled features anyway.